pkg/api/v1/types.go

```diff
@@ -1649,6 +1649,8 @@ type Endpoints struct {
 type EndpointSubset struct {
 	// IP addresses which offer the related ports.
 	Addresses []EndpointAddress `json:"addresses,omitempty"`
+	// IP addresses which offer the related ports but have not yet passed readiness checks
+	UnreadyAddresses []EndpointAddress `json:"unreadyAddresses,omitempty"`
```

Comment:
Yeah, my primary concern was not upsetting the core path but still enabling the access.